server/src/providers/project/index.ts

```diff
@@ -40,34 +45,60 @@ export async function initialise() {
 }
 
 async function loadWorkspace() {
-	const uris = await getProjectFiles();
+	const workspaces = await connection.workspace.getWorkspaceFolders();
+
+	console.log(workspaces);
+
+	if (workspaces) {
+		let uris: string[] = [];
+
+		workspaces.forEach((workspaceUri => {
+			const folderPath = URI.parse(workspaceUri.uri).path;
+
+			console.log(`Starting search of: ${folderPath}`);
+			const files = glob.sync(projectFilesGlob, {
+				cwd: folderPath,
+				absolute: true,
+				nocase: true,
+			});
+
+			console.log(`Found RPGLE files: ${files.length}`);
 
-	if (uris) {
-		await Promise.allSettled(uris?.map(uri => loadFile(uri)));
+			uris.push(...files.map(file => URI.from({
+				scheme: `file`,
+				path: file
+			}).toString()))
+		}));
+
+		if (uris.length < 1000) {
```

Comment:
For the time being, we will disable project mode if more than 1000 sources are found.